### ISSUE TITLE
rendering of alt attribute should be configurable

### ISSUE THREAD

**Issue by tcurdt** [arve0/markdown-it-implicit-figures#39]
As discussed in #38 and based on https://www.scottohara.me/blog/2019/01/21/how-do-you-figure.html#a-figcaption-is-not-a-substitute-for-alt-text it should be configurable whether the alt attribute should be rendered or kept empty.

How about:

```
md.use(implicitFigures, {
  dataType: false,  // <figure data-type="image">, default: false
  figcaption: false,  // <figcaption>alternative text</figcaption>, default: false
  alt: true, // <img alt="alternative text" />, default: true
  tabindex: false, // <figure tabindex="1+n">..., default: false
  link: false // <a href="img.png"><img src="img.png"></a>, default: false
});
```

**Comment by tcurdt** [comment]
I just looked at the code to see if could come up with a PR but I am a little confused. Looking at https://github.com/arve0/markdown-it-implicit-figures/blob/master/index.js#L60 - what is actually setting the `alt` attribute to empty? I don't see it.

**Comment by arve0** [comment]
It’s not clear, no, but use git blame and the commit log and you will find this: https://github.com/arve0/markdown-it-implicit-figures/commit/cd5141df41f4f3e0cff220d7c61dd6d3d8038171

**Comment by tcurdt** [comment]
@arve0 of course that's what I did - but `image.children.length = 0` does not make it any more clear.

If attributes are modeled as children, the image would have no attributes anymore - but it needs to keep the `src`. And the `alt` is empty and not gone.

If attributes are modeled on the node, the image would not have any children anyway.

Hence my confusion.

**Comment by arve0** [comment]
Sorry, did not mean to offend you in any way. It’s not not clear to me either. Have you tried debugging and stepping through it?

**Comment by tcurdt** [comment]
Sorry, I did not mean to give that impression. Ups! All good on my side.

No, I didn't step through yet. For now I was just looking at the code and trying to make sense of it :)

**Comment by sajozsattila** [comment]
The https://github.com/arve0/markdown-it-implicit-figures/blob/master/index.js#L69:

`image.children.length = 0;`

makes the alt is null. However, I do not really  understand why is there. At my local version, I have just commented out this line, and so far I do not experience any problem. Anybody know why this line was inserted in first place?

**Comment by changethe** [comment]
any update on this? would be good to be able to keep the alt text.

**Comment by arve0** [comment]
This should be fixed in v0.12.0.

Thank you @ItsEricWoodward.
